[Flaky Test] TestFilestreamTruncate

## Flaky Test

* **Test Name:** TestFilestreamTruncate
* **Link:** https://github.com/elastic/beats/blob/a9279cd12db9a4ffc0b0685ee613faf4c7771508/filebeat/input/filestream/input_integration_test.go#L862
* **Branch:** master

Looks like it was recently introduced in #24425 and has been failing on and off since then.

<img width="1357" alt="Screen Shot 2021-04-21 at 5 15 54 PM" src="https://user-images.githubusercontent.com/4565752/115622175-821b0d00-a2c5-11eb-8c3f-b92fc5b83013.png">


### Stack Trace

```
=== RUN   TestFilestreamTruncate
coverage: 67.6% of statements
panic: test timed out after 10m0s

goroutine 343 [running]:
testing.(*M).startAlarm.func1()
	/usr/local/go/src/testing/testing.go:1618 +0xe5
created by time.goFunc
	/usr/local/go/src/time/sleep.go:167 +0x45

goroutine 1 [chan receive, 9 minutes]:
testing.(*T).Run(0xc000682600, 0xd1bbdb, 0x16, 0xd42618, 0x495101)
	/usr/local/go/src/testing/testing.go:1169 +0x2da
testing.runTests.func1(0xc0001ee900)
	/usr/local/go/src/testing/testing.go:1439 +0x78
testing.tRunner(0xc0001ee900, 0xc00026fda8)
	/usr/local/go/src/testing/testing.go:1123 +0xef
testing.runTests(0xc0004a5400, 0x134aee0, 0x23, 0x23, 0xc0183ff58dbfd63b, 0x8bb36d26c0, 0x136e160, 0x40f470)
	/usr/local/go/src/testing/testing.go:1437 +0x2fe
testing.(*M).Run(0xc0000dd500, 0x0)
	/usr/local/go/src/testing/testing.go:1345 +0x1eb
main.main()
	_testmain.go:171 +0x1c5

goroutine 386 [runnable]:
github.com/elastic/go-structform/gotype.foldMapInterface(0xc00052a180, 0xc12500, 0xc0001f0ff0, 0x13a0760, 0xc000296000)
	/go/pkg/mod/github.com/elastic/go-structform@v0.0.8/gotype/fold_map.go:51 +0xdc
github.com/elastic/go-structform/gotype.foldInterfaceValue(0xc00052a180, 0xc12500, 0xc0001f0ff0, 0x0, 0x0)
	/go/pkg/mod/github.com/elastic/go-structform@v0.0.8/gotype/fold.go:98 +0x445
github.com/elastic/go-structform/gotype.foldMapInterface(0xc00052a180, 0xc12500, 0xc0001f0f90, 0x13a0760, 0xbb0080)
	/go/pkg/mod/github.com/elastic/go-structform@v0.0.8/gotype/fold_map.go:55 +0x171
github.com/elastic/go-structform/gotype.foldInterfaceValue(0xc00052a180, 0xc12500, 0xc0001f0f90, 0x0, 0x0)
	/go/pkg/mod/github.com/elastic/go-structform@v0.0.8/gotype/fold.go:98 +0x445
github.com/elastic/go-structform/gotype.(*Iterator).Fold(...)
	/go/pkg/mod/github.com/elastic/go-structform@v0.0.8/gotype/fold.go:86
github.com/elastic/beats/v7/libbeat/common/transform/typeconv.(*Converter).Convert(0xc000528000, 0xbb0080, 0xc0003a49a0, 0xc12500, 0xc0001f0f90, 0x0, 0x0)
	/go/src/github.com/elastic/beats/libbeat/common/transform/typeconv/typeconv.go:108 +0x136
github.com/elastic/beats/v7/libbeat/common/transform/typeconv.Convert(0xbb0080, 0xc0003a49a0, 0xc12500, 0xc0001f0f90, 0x0, 0x0)
	/go/src/github.com/elastic/beats/libbeat/common/transform/typeconv/typeconv.go:129 +0xc7
github.com/elastic/beats/v7/libbeat/statestore/storetest.(*MapStore).Get(0xc00019db00, 0xc00003ea20, 0x28, 0xbb0080, 0xc0003a49a0, 0x0, 0x0)
	/go/src/github.com/elastic/beats/libbeat/statestore/storetest/storetest.go:156 +0xe5
github.com/elastic/beats/v7/libbeat/statestore.(*Store).Get(0xc0003a1c40, 0xc00003ea20, 0x28, 0xbb0080, 0xc0003a49a0, 0x0, 0x0)
	/go/src/github.com/elastic/beats/libbeat/statestore/store.go:101 +0x13c
github.com/elastic/beats/v7/filebeat/input/filestream.(*inputTestingEnvironment).getRegistryState(0xc00049e1e0, 0xc00003ea20, 0x28, 0x22, 0xc12500, 0xc0003a6db0, 0x0, 0x0)
	/go/src/github.com/elastic/beats/filebeat/input/filestream/environment_test.go:275 +0x98
github.com/elastic/beats/v7/filebeat/input/filestream.(*inputTestingEnvironment).waitUntilOffsetInRegistry(0xc00049e1e0, 0xd11678, 0x8, 0x0)
	/go/src/github.com/elastic/beats/filebeat/input/filestream/environment_test.go:238 +0x1ed
github.com/elastic/beats/v7/filebeat/input/filestream.TestFilestreamTruncate(0xc000682600)
	/go/src/github.com/elastic/beats/filebeat/input/filestream/input_integration_test.go:890 +0x485
testing.tRunner(0xc000682600, 0xd42618)
	/usr/local/go/src/testing/testing.go:1123 +0xef
created by testing.(*T).Run
	/usr/local/go/src/testing/testing.go:1168 +0x2b3

goroutine 390 [semacquire, 9 minutes]:
sync.runtime_Semacquire(0xc0001f09b8)
	/usr/local/go/src/runtime/sema.go:56 +0x45
sync.(*WaitGroup).Wait(0xc0001f09b0)
	/usr/local/go/src/sync/waitgroup.go:130 +0x65
github.com/elastic/go-concert/unison.(*MultiErrGroup).Wait(0xc0001f0990, 0x0, 0x0, 0x0)
	/go/pkg/mod/github.com/elastic/go-concert@v0.1.0/unison/multierrgroup.go:54 +0x53
github.com/elastic/beats/v7/filebeat/input/filestream.(*fileProspector).Run(0xc000172280, 0xc00006eff0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/go/src/github.com/elastic/beats/filebeat/input/filestream/prospector.go:229 +0x3a5
github.com/elastic/beats/v7/filebeat/input/filestream/internal/input-logfile.(*managedInput).Run(0xc0001722d0, 0xc00006eff0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/go/src/github.com/elastic/beats/filebeat/input/filestream/internal/input-logfile/input.go:78 +0x38d
github.com/elastic/beats/v7/filebeat/input/filestream.(*inputTestingEnvironment).startInput.func1(0xe01b80, 0xc0002913c0, 0xdff040, 0xc0001722d0, 0xc00049e1e0, 0xc00049e270)
	/go/src/github.com/elastic/beats/filebeat/input/filestream/environment_test.go:95 +0x14c
created by github.com/elastic/beats/v7/filebeat/input/filestream.(*inputTestingEnvironment).startInput
	/go/src/github.com/elastic/beats/filebeat/input/filestream/environment_test.go:91 +0x9a

goroutine 391 [select]:
github.com/elastic/go-concert/timed.Periodic(0x7f71fa899030, 0xc000291400, 0xf4240, 0xc0004e5ec0, 0x0, 0x0)
	/go/pkg/mod/github.com/elastic/go-concert@v0.1.0/timed/timed.go:76 +0x13d
github.com/elastic/beats/v7/filebeat/input/filestream.(*fileWatcher).Run(0xc000291380, 0x7f71fa899008, 0xc000291400)
	/go/src/github.com/elastic/beats/filebeat/input/filestream/fswatch.go:128 +0x10b
github.com/elastic/beats/v7/filebeat/input/filestream.(*fileProspector).Run.func1(0x0, 0x0)
	/go/src/github.com/elastic/beats/filebeat/input/filestream/prospector.go:117 +0x9a
github.com/elastic/go-concert/unison.(*MultiErrGroup).Go.func1(0xc0001f0990, 0xc0001aa920)
	/go/pkg/mod/github.com/elastic/go-concert@v0.1.0/unison/multierrgroup.go:42 +0x64
created by github.com/elastic/go-concert/unison.(*MultiErrGroup).Go
	/go/pkg/mod/github.com/elastic/go-concert@v0.1.0/unison/multierrgroup.go:40 +0x66

goroutine 392 [chan receive, 9 minutes]:
github.com/elastic/beats/v7/filebeat/input/filestream.(*fileWatcher).Event(0xc000291380, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
	/go/src/github.com/elastic/beats/filebeat/input/filestream/fswatch.go:228 +0x85
github.com/elastic/beats/v7/filebeat/input/filestream.(*fileProspector).Run.func2(0xc000342ea0, 0x0)
	/go/src/github.com/elastic/beats/filebeat/input/filestream/prospector.go:125 +0x1bf
github.com/elastic/go-concert/unison.(*MultiErrGroup).Go.func1(0xc0001f0990, 0xc000291440)
	/go/pkg/mod/github.com/elastic/go-concert@v0.1.0/unison/multierrgroup.go:42 +0x64
created by github.com/elastic/go-concert/unison.(*MultiErrGroup).Go
	/go/pkg/mod/github.com/elastic/go-concert@v0.1.0/unison/multierrgroup.go:40 +0x66
```

--- a/filebeat/input/filestream/input_integration_test.go
+++ b/filebeat/input/filestream/input_integration_test.go
@@ -860,8 +860,6 @@ func TestFilestreamSymlinkRemoved(t *testing.T) {
 
 // test_truncate from test_harvester.py
 func TestFilestreamTruncate(t *testing.T) {
-	t.Skip("Flaky test: https://github.com/elastic/beats/issues/25217")
-
 	env := newInputTestingEnvironment(t)
 
 	testlogName := "test.log"
@@ -870,8 +868,9 @@ func TestFilestreamTruncate(t *testing.T) {
 		"paths": []string{
 			env.abspath("*"),
 		},
-		"prospector.scanner.check_interval": "1ms",
-		"prospector.scanner.symlinks":       "true",
+		"prospector.scanner.check_interval":  "1ms",
+		"prospector.scanner.resend_on_touch": "true",
+		"prospector.scanner.symlinks":        "true",
 	})
 
 	lines := []byte("first line\nsecond line\nthird line\n")
@@ -897,8 +896,7 @@ func TestFilestreamTruncate(t *testing.T) {
 	moreLines := []byte("forth line\nfifth line\n")
 	env.mustWriteLinesToFile(testlogName, moreLines)
 
-	env.waitUntilEventCount(5)
-	env.requireOffsetInRegistry(testlogName, len(moreLines))
+	env.waitUntilOffsetInRegistry(testlogName, len(moreLines))
 
 	cancelInput()
 	env.waitUntilInputStops()

